source/automated-deployment.rst

```diff
@@ -0,0 +1,84 @@
+Automating Deployment with Puppet
+=================================
+Definitions
+-----------
+* BEAMLINE_WORKSTATION
+    Any computer that beamline staff designate as a computer they want the
+    collection stack installed on
+* BEAMINE_SERVER
+    The main "compute" computer where jupyterhub will point to
+* IPYTHON_CONFIGURATION:
```

Comment:
That is definitely true.  My experience was that most beamlines have trouble keeping all of their files checked in.  I did a quick investigation of some of the beamlines and found that many of them had files that were not being tracked but were being run at startup. That info is [here] (https://gist.github.com/ericdill/6bbfb85e9db2ca8b4d00d53c6252bf66).  

There are a number of ways to solve this problem.

1) What I propose here (simply copying the folder to somewhere else)

```bash
cp /home/beamline_user/ipython_profile /path/to/backup/ipython_profile_YYYY-Week -r
```
2) Automatically commit files that are being tracked:
```bash
pushd /home/beamline_user/ipython_profile
git commit --allow-empty -am "Automated weekly code backup. YYYY-Week"
git tag -a YYYY-Week -m "Automated weekly tag: YYYY-Week"
popd
```
3) Commit *all* python files and add any files that are already being tracked

```bash
pushd /home/beamilne_user/ipython_profile 
find -name *.py -exec git add {} \; 
git commit -am "Automated weekly code backup: YYYY-Week"
git tag -a YYYY-Week -m "Automated weekly tag: YYYY-Week"
popd
```

In either case (copying or auto-committing + tagging) this would serve as a marker in time where it would be reasonably straightforward to roll back to a previous known working state.

```bash
source activate collection_2016-05
cd /path/to/ipython/profile
git checkout 2016-05
ipython --profile=collection_2016-05
```

Do you like any of these options?